lib/rabbitmq/cli/command_behaviour.ex

```diff
@@ -29,7 +29,9 @@ defmodule RabbitMQ.CLI.CommandBehaviour do
                       scopes: 0,
                       usage_additional: 0,
                       switches: 0,
-                      aliases: 0
+                      aliases: 0,
+                      required_rabbit_app_state: 2,
+                      validate_execution_environment: 2
```

Comment:
How useful is `rabbitmq-diagnostics memory_breakdown` on a VM with `rabbit` not running? We are not trying to reinvent Erlang tooling after all.

Commands that are ambivalent about the state of the `rabbit` app either operate on an unused data directory (`rename_cluster_node`) or Erlang VM. There's only a handful of them. Are they worth going through every single command and adding a validation function plus requiring every other command that exists or will be added to be updated the same way?

If we fold both validations into `validate_execution_environment/2`, what should happen when it is not defined? No validations should be performed at all?